File: android/src/main/java-mapboxgl/common/com/mapbox/rctmgl/components/AbstractEvent.java
package com.mapbox.rctmgl.components;

import com.facebook.react.bridge.WritableMap;
import com.facebook.react.uimanager.events.Event;
import com.facebook.react.uimanager.events.RCTEventEmitter;

import javax.annotation.Nullable;

public class AbstractEvent extends Event<AbstractEvent> {
    private String mEventName;
    private final boolean mCanCoalesce;
    private WritableMap mEvent;

    public AbstractEvent(int viewId, String eventName, boolean canCoalesce, @Nullable WritableMap event) {
        super(viewId);
        mEventName = eventName;
        mCanCoalesce = canCoalesce;
        mEvent = event;
    }

    @Override
    public String getEventName() {
        return mEventName;
    }

    @Override
    public void dispatch(RCTEventEmitter rctEventEmitter) {
        rctEventEmitter.receiveEvent(getViewTag(), getEventName(), mEvent);
    }

    @Override
    public boolean canCoalesce() {
        return mCanCoalesce;
    }
}

File: android/src/main/java/com/rnmapbox/rnmbx/components/AbstractEvent.kt
package com.rnmapbox.rnmbx.components

import com.facebook.react.bridge.WritableMap
import com.facebook.react.uimanager.events.Event
import com.facebook.react.uimanager.events.RCTEventEmitter

class AbstractEvent(
    viewId: Int,
    private val mEventName: String,
    private val mCanCoalesce: Boolean,
    private val mEvent: WritableMap?
) : Event<AbstractEvent>(viewId) {
    override fun getEventName(): String {
        return mEventName
    }

    override fun dispatch(rctEventEmitter: RCTEventEmitter) {
        rctEventEmitter.receiveEvent(viewTag, eventName, mEvent)
    }

    override fun canCoalesce(): Boolean {
        return mCanCoalesce
    }
}